src/node_watchdog.h

```diff
@@ -55,24 +56,53 @@ class Watchdog {
   bool* timed_out_;
 };
 
-class SigintWatchdog {
+class SigintWatchdogBase {
+ public:
+  virtual bool HandleSigint() {
```

Comment:
```suggestion
  virtual ~SigintWatchdogBase() {}
  virtual bool HandleSigint() = 0;
```

The virtual destructor is a best practice for base classes, and using `= 0` forces subclasses to override the method at compile time.